wasm/src/debug.rs

```diff
@@ -61,11 +78,74 @@ impl<T> DebugUnwrapExt for Option<T> {
 
     #[allow(clippy::panic)]
     fn dbg_unwrap(self) -> Self::Output {
-        match self {
-            Some(out) => out,
-            None => {
-                dbg("WASM execution panicked at 'called `Option::dbg_unwrap()` on a `None` value'");
-                panic!("");
+        #[cfg(not(feature = "debug"))]
+        #[allow(clippy::unwrap_used)]
+        return self.unwrap();
+
+        #[cfg(feature = "debug")]
+        {
+            match self {
+                Some(out) => out,
+                None => {
+                    dbg("WASM execution panicked at 'called `Option::dbg_unwrap()` on a `None` value'");
+                    panic!("");
+                }
+            }
+        }
+    }
+}
+
+/// Extension implemented for `Result` and `Option` to provide expecting with error message,
+/// cause basic `expect()` does not print error due to specific panic handling in WASM Runtime.
+///
+/// Expands to just `expect()` if `debug` feature is not specified
+pub trait DebugExpectExt {
+    /// Type of the value that is returned in success
+    type Output;
+
+    /// Just like `expect()` but prints error message before panic
+    fn dbg_expect(self, msg: &str) -> Self::Output;
+}
+
+impl<T, E: Debug> DebugExpectExt for Result<T, E> {
+    type Output = T;
+
+    #[allow(clippy::panic)]
+    fn dbg_expect(self, msg: &str) -> Self::Output {
```

Comment:
Hm..., I'm looking at it more like [`colored crate`](https://crates.io/crates/colored) rather than like `proc_macro_error`.

With `colored` you can write code like `println!("foo".blue());`. This will display `foo` using blue color. But if you enable feature `no-color`, then this `blue()` will do nothing. This is usefull, because you can have beautiful output for normal terminals and have optional feature `dumb_terminal = ["colored/no-color"]` for terminals without color support.

The same thing here. I'd like the users to keep their code the same and only manipulate with feature flag to enable/disable debugging.

`expect_or_abort` has different meaning -- it will always abort on `Err/None` variant.

Reply:
Let's summon more voices here as well